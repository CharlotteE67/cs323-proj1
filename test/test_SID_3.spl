struct test{
    int a;
    char b;
};
int test_func(char c){
    float a,b;
    while(a==b){
        if(c>=2){
            a = 5;
        }else{
            b = 6;
        }
    }
    return a==b;
}
int main(){

    struct test my_test;
    int d,final;
    if(d!=0){
        if(a&&b){
            final = test_func(my_test.a);
        }
        
    }else{
        final = test_func(0);
    }

}