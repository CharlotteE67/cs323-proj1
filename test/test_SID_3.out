Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        StructSpecifier (1)
          STRUCT
          ID: test
          LC
          DefList (2)
            Def (2)
              Specifier (2)
                TYPE: int
              DecList (2)
                Dec (2)
                  VarDec (2)
                    ID: a
              SEMI
            DefList (3)
              Def (3)
                Specifier (3)
                  TYPE: char
                DecList (3)
                  Dec (3)
                    VarDec (3)
                      ID: b
                SEMI
          RC
      SEMI
    ExtDefList (5)
      ExtDef (5)
        Specifier (5)
          TYPE: int
        FunDec (5)
          ID: test_func
          LP
          VarList (5)
            ParamDec (5)
              Specifier (5)
                TYPE: char
              VarDec (5)
                ID: c
          RP
        CompSt (5)
          LC
          DefList (6)
            Def (6)
              Specifier (6)
                TYPE: float
              DecList (6)
                Dec (6)
                  VarDec (6)
                    ID: a
                COMMA
                DecList (6)
                  Dec (6)
                    VarDec (6)
                      ID: b
              SEMI
          StmtList (7)
            Stmt (7)
              WHILE
              LP
              Exp (7)
                Exp (7)
                  ID: a
                EQ
                Exp (7)
                  ID: b
              RP
              Stmt (7)
                CompSt (7)
                  LC
                  StmtList (8)
                    Stmt (8)
                      IF
                      LP
                      Exp (8)
                        Exp (8)
                          ID: c
                        GE
                        Exp (8)
                          INT: 2
                      RP
                      Stmt (8)
                        CompSt (8)
                          LC
                          StmtList (9)
                            Stmt (9)
                              Exp (9)
                                Exp (9)
                                  ID: a
                                ASSIGN
                                Exp (9)
                                  INT: 5
                              SEMI
                          RC
                      ELSE
                      Stmt (10)
                        CompSt (10)
                          LC
                          StmtList (11)
                            Stmt (11)
                              Exp (11)
                                Exp (11)
                                  ID: b
                                ASSIGN
                                Exp (11)
                                  INT: 6
                              SEMI
                          RC
                  RC
            StmtList (14)
              Stmt (14)
                RETURN
                Exp (14)
                  Exp (14)
                    ID: a
                  EQ
                  Exp (14)
                    ID: b
                SEMI
          RC
      ExtDefList (16)
        ExtDef (16)
          Specifier (16)
            TYPE: int
          FunDec (16)
            ID: main
            LP
            RP
          CompSt (16)
            LC
            DefList (18)
              Def (18)
                Specifier (18)
                  StructSpecifier (18)
                    STRUCT
                    ID: test
                DecList (18)
                  Dec (18)
                    VarDec (18)
                      ID: my_test
                SEMI
              DefList (19)
                Def (19)
                  Specifier (19)
                    TYPE: int
                  DecList (19)
                    Dec (19)
                      VarDec (19)
                        ID: d
                    COMMA
                    DecList (19)
                      Dec (19)
                        VarDec (19)
                          ID: final
                  SEMI
            StmtList (20)
              Stmt (20)
                IF
                LP
                Exp (20)
                  Exp (20)
                    ID: d
                  NE
                  Exp (20)
                    INT: 0
                RP
                Stmt (20)
                  CompSt (20)
                    LC
                    StmtList (21)
                      Stmt (21)
                        IF
                        LP
                        Exp (21)
                          Exp (21)
                            ID: a
                          AND
                          Exp (21)
                            ID: b
                        RP
                        Stmt (21)
                          CompSt (21)
                            LC
                            StmtList (22)
                              Stmt (22)
                                Exp (22)
                                  Exp (22)
                                    ID: final
                                  ASSIGN
                                  Exp (22)
                                    ID: test_func
                                    LP
                                    Args (22)
                                      Exp (22)
                                        Exp (22)
                                          ID: my_test
                                        DOT
                                        ID: a
                                    RP
                                SEMI
                            RC
                    RC
                ELSE
                Stmt (25)
                  CompSt (25)
                    LC
                    StmtList (26)
                      Stmt (26)
                        Exp (26)
                          Exp (26)
                            ID: final
                          ASSIGN
                          Exp (26)
                            ID: test_func
                            LP
                            Args (26)
                              Exp (26)
                                INT: 0
                            RP
                        SEMI
                    RC
            RC
